django/db/migrations/autodetector.py

```diff
@@ -576,6 +591,134 @@ def generate_renamed_models(self):
+                        )
+                        model = self.to_state.apps.get_model(app_label, model_name)
+                        fields = model._meta.get_fields(include_hidden=False)
+                        dependencies = list(dependencies) + [
+                            (app_label, model_name, None, True),
+                        ]
+                        alter_field_deps = []
+
+                        for field in fields:
+                            if (
+                                field.is_relation
+                                and field.remote_field.related_model != model
+                            ):
+                                self.add_operation(
+                                    field.related_model._meta.model_name,
+                                    operations.AlterField(
+                                        model_name=field.related_model._meta.model_name,
+                                        name=field.remote_field.name,
+                                        field=field.remote_field,
+                                    ),
+                                    dependencies=dependencies,
+                                )
+                                self.already_alter_fields.add(
+                                    (
+                                        field.related_model._meta.model_name,
+                                        field.related_model._meta.model_name,
+                                        field.remote_field.name,
+                                    )
+                                )
+                                alter_field_deps.append(
+                                    (
+                                        rem_app_label,
+                                        field.related_model._meta.model_name,
+                                        field.remote_field.name,
+                                        "alter",
+                                    ),
+                                )
+                        dependencies = [
+                            op for op in chain(dependencies, alter_field_deps)
+                        ]
+
+                        self.add_operation(
+                            rem_app_label,
+                            operations.AlterModelOptions(
+                                name=rem_model_name, options={"managed": False}
+                            ),
+                        )
+
+                        self.add_operation(
+                            rem_app_label,
+                            operations.DeleteModel(name=rem_model_name),
+                            dependencies=dependencies,
+                        )
+                        self.old_model_keys.remove((rem_app_label, rem_model_name))
+                        self.old_model_keys.add((app_label, model_name))
+                        self.from_state.add_model(model_state)
+                        break
```

Comment:
This shows, in part, why long functions are problematic: My first reaction to this line was, "WTF?". I had to look more than 100 lines up to see the context that the `break` refers to.

And actually, the line is dubious -- it says "after we found a new model which matches the old model, and the user acknowledged the move, we don't need to look for more matching models". I think it would be more friendly to keep looking, and alert them if there are more matching targets for the move -- although this can be left for later, after we get a first working version.